app/controllers/concerns/sessions_controller_override.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module SessionsControllerOverride
+  def create
+    super do |user|
+      verify_users_after_sign_in(user)
+    end
+  end
+
+  private
+
+  def verify_users_after_sign_in(user)
+    record = Decidim::Verifications::CsvDatum.find_or_create_by(email: user.email, organization: user.organization)
+    return unless record
+
+    authorization = Decidim::Authorization.find_or_initialize_by(
+      user:,
+      name: "csv_census"
+    )
+
```

Comment:
This code does not take into account that the authorization method is registered in the organization.

You can use the `current_organization.available_authorizations` for that.